added replicated-rbac, basic schema

jgruica opened the pull request:
- [ ] If any changes have been made to `/project` files, the corresponding `/src` files have been regenerated.
- [ ] If any changes need to be deployed, the version has been bumped in `package.json`.
- [ ] If any changes impact the replicated-lint package, they have been added to `CHANGELOG`.

jgruica committed:
added replicated-rbac, basic schema

dexhorthy commented on src/schemas/replicated.json:
```diff
@@ -0,0 +1,657 @@
+{
```
Where did this file come from? Was this committed by accident?

jgruica replied:
I didn't touch that file. How can I get rid of it? Should I just delete that?

dexhorthy commented on projects/replicated-rbac/rules.yaml:
```diff
@@ -0,0 +1,27 @@
+---
```
Is this rule necessary? I would think this is enforced by `required: ["name" ...]` in the schema

jgruica replied:
because we are just verifying shape of the policy i don't think so that we need rules.yaml file. but when i'm trying to remove that i'm getting some error. so i need to figure out what to do.

dexhorthy replied:
You might be able to just set rules.yaml to an empty list:

```
---
[]
```

jgruica committed:
added empty array to rules yaml

jgruica committed:
custom rules for invalid policy

jgruica commented:
@dexhorthy i modified predictates.ts
Before was ```const value = _.get(object, this.path);``` and i added ```const value = this.path ? _.get(object, this.path) : object;```
This is maybe permanent solution and we can talk about that tomorrow.
Problem was that allowed and denied are arrays of strings (primitive values), and get from lodash is expecting an object but not "primitives".

dexhorthy commented on src/predicates.ts:
```diff
@@ -63,7 +63,9 @@ export class NotMatch implements Predicate<any> {
   public test(object: any): RuleMatchedAt {
-    const value = _.get(object, this.path);
+    // const value = _.get(object, this.path);
+    const value = this.path ? _.get(object, this.path) : object;
```
this makes sense to me :+1:

jgruica committed:
removed replicated.json

jgruica committed:
removed commented code, added correct link to policies

jgruica committed:
added link for denied policy

jgruica committed:
policy message
